From 87dc89966c3d36cffe63c25bd875aac4d6364bf2 Mon Sep 17 00:00:00 2001
From: Wenping Zhang <wenping.zhang@rock-chips.com>
Date: Tue, 25 Sep 2018 15:54:10 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3399: change the parameter of
 vdd_log.

There is a hardware mistake on firefly board, it will cause the
vdd_log is too low based on previous parameter, so we change the
pwm parameter to make sure the vdd_log is about 0.9v.

Change-Id: I2d63cfe201c02a1351eebae21d1c7d449385087e
Signed-off-by: Wenping Zhang <wenping.zhang@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts | 9 ++++++++-
 arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts   | 9 ++++++++-
 2 files changed, 16 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
index 2859986d630c..a8294924e17b 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
@@ -338,7 +338,14 @@
 		pwms = <&pwm2 0 25000 1>;
 		regulator-name = "vdd_log";
 		regulator-min-microvolt = <800000>;
-		regulator-max-microvolt = <1400000>;
+		/*
+		 * the firefly hardware using 3.0 v as APIO2_VDD
+		 * voltage, but the pwm divider resistance is designed
+		 * based on hardware which the APIO2_VDD is 1.8v, so we
+		 * need to change the regulator-max-microvolt from 1.4v
+		 * to 1.0v, so the pwm can output 0.9v voltage.
+		 */
+		regulator-max-microvolt = <1000000>;
 		regulator-always-on;
 		regulator-boot-on;
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
index 99d40a3f7115..82bb2a63fe73 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
@@ -287,7 +287,14 @@
 		pwms = <&pwm2 0 25000 1>;
 		regulator-name = "vdd_log";
 		regulator-min-microvolt = <800000>;
-		regulator-max-microvolt = <1400000>;
+		/*
+		 * the firefly hardware using 3.0 v as APIO2_VDD
+		 * voltage, but the pwm divider resistance is designed
+		 * based on hardware which the APIO2_VDD is 1.8v, so we
+		 * need to change the regulator-max-microvolt from 1.4v
+		 * to 1.0v, so the pwm can output 0.9v voltage.
+		 */
+		regulator-max-microvolt = <1000000>;
 		regulator-always-on;
 		regulator-boot-on;
 
-- 
2.35.3

